dataBase.c

```diff
@@ -0,0 +1,130 @@
+#include "dataBase.h"
+#include "school.h"
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+#define _CRT_SECURE_NO_WARNINGS
+#define MAX_LINE_LENGTH 150
+StudentNode* g_schooldDataBase[NUM_OF_LEVELS][NUM_OF_CLASSES];
+static Student* _parseLine(char* line, int* sClass, int* level)
+{
+    char* token;
+    if (line[strlen(line) - 1] == '\n') {
+        line[strlen(line) - 1] = '\0';
+    }
+
+    Student* newStudent = (Student*)malloc(sizeof(Student));
+    if (!newStudent) {
+        return NULL;
+    }
+    newStudent->firstName = (char*)malloc(sizeof(Student));
+    token = strtok(line, " ");
+    strncpy(newStudent->firstName, token, sizeof(token) - 1);
+    newStudent->firstName[sizeof(newStudent->firstName) - 1] = '\0';
```

Comment:
firstName is a pointer, then size will be 7 bytes?
You have a misunderstanding regartding sizeof function in C.
Read about it, in addition to size of variables.